Updating Hidden File detector for SVN

sgerlach opened the pull request:
Signed-off-by: Scott Gerlach <scott.gerlach@stackhawk.com>

Resolved some false positives for the .svn/entries detector triggering on HTML content AND adds the new wc.db file for newer versions of SVN.

sgerlach commented:
Hopefully this is in good shape. Pulled it together based on some feedback I saw and wanted to make this more accurate.

kingthorin commented on addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/HiddenFilesScanRuleUnitTest.java:
```diff
@@ -478,6 +478,140 @@ void shouldRaiseAlertIfTestedUrlRespondsOkWithRelevantContentAndAppropriateNotCo
+
+        // When
+        rule.scan();
+        // Then
+        // Then
```
Drop one?

kingthorin commented on addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/HiddenFilesScanRuleUnitTest.java:
```diff
@@ -478,6 +478,140 @@ void shouldRaiseAlertIfTestedUrlRespondsOkWithRelevantContentAndAppropriateNotCo
     }
 
+    @Test
+    void
+    shouldRaiseAlertIfTestedUrlRespondsOkWithRelevantContentSVNEntries()
```
I think these can be put on a single line, the one before is really long hence it being wrapped.

kingthorin commented on addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/HiddenFilesScanRuleUnitTest.java:
```diff
@@ -478,6 +478,140 @@ void shouldRaiseAlertIfTestedUrlRespondsOkWithRelevantContentAndAppropriateNotCo
+
+        // When
+        rule.scan();
+        // Then
+        // Then
```
Drop one?

kingthorin commented:
Could you please run the `spotlessApply` gradle task? (do an ammending commit and force push)

sgerlach commented:
I knew I forgot something!!! Thanks @kingthorin

kingthorin commented on addOns/ascanrulesBeta/CHANGELOG.md:
```diff
@@ -37,6 +37,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
     - The URI field will be in encoded form.
   - XPath Injection
     - Added evidence
+  - Hidden File
+    - Some content checking has been added for .svn/entries as well as detection for wc.db
```
It seems like this PR was started from a pretty old base. You might want to revert this, fetch upstream, rebase then re-add the change note in the current "Unreleased" section. (Note: Changes aren't usually two lines, the case you added to here was more of a one off ... have  a look at the entries in the up-to-date file and you'll see what I mean)

kingthorin reviewed (commented):
Sorry I didn't notice this earlier.

kingthorin commented:
@sgerlach let me know if you need me to tackle it, I know this can be finicky.

sgerlach commented:
I think I got it

thc202 commented:
Needs the rebase? It still has conflicts.

sgerlach commented:
@thc202 I def need help here. The only conflict I see in that file is the update I added (after I rebased and re-added the update). As far as I can tell I did this the right way, but now I'm lost

kingthorin commented:
I got it. Will update the PR shortly.

kingthorin commented:
Nice work 😀

thc202 commented on addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/HiddenFilesScanRuleUnitTest.java:
```diff
@@ -481,6 +481,132 @@ void shouldRaiseAlertIfTestedUrlRespondsOkWithRelevantContentAndAppropriateNotCo
     }
 
+    @Test
+    void shouldRaiseAlertIfTestedUrlRespondsOkWithRelevantContentSVNEntries()
```
Are these tests really needed? They seem to be testing what was already tested earlier (it might not be the same content but the logic being tested is the same).

sgerlach replied:
Heh - just trying to make sure the check triggers less false positives on updates, but I can go either way on the "need"

sgerlach replied:
hmmm, how tricky would it be to make them use the file in the test? If it's not hard, I might be up for changing that, else I'm happy to remove.

sgerlach replied:
removed these

thc202 replied:
They are still present in the latest commit.

Regarding the previous question, I  think it just needs to call `init()` to load the file.

sgerlach replied:
Heh, I missed the git add. should be correct now

thc202 replied:
There are still some?

There's still a commit pending above: https://github.com/zaproxy/zap-extensions/pull/3471#discussion_r788968874

thc202 replied:
It's now under :)

kingthorin commented on addOns/ascanrulesBeta/CHANGELOG.md:
```diff
@@ -4,7 +4,8 @@ All notable changes to this add-on will be documented in this file.
 ## Unreleased
-
+### Changed
+- Hidden File added some content checking has been added for .svn/entries as well as detection for wc.db
```
```suggestion

- Hidden File Finder scan rule, content checking has been added for .svn/entries as well as detection for wc.db.
```

thc202 commented:
Thank you!

kingthorin commented on addOns/ascanrulesBeta/src/main/zapHomeFiles/json/hidden_files.json:
```diff
@@ -35,6 +35,14 @@
+      "content":["FILE","DIR"],
+      "links":["https://www.oreilly.com/library/view/version-control-with/9780596510336/ch08s02s01.html"],
+      "type":"svn_dir",
+      "source":"snallygaster"
```
@sgerlach Are these really both from snallygaster?

sgerlach replied:
Actually information I gathered from these resources as well as snallygaster and some local research I did
https://github.com/NetSPI/SVNentriesParser
https://cirt.net/svnpristine

kingthorin replied:
Okay, I was trying to ensure we attributed things properly. I guess the .svn/entries one is originally from snallygaster though You've improved it here, and the .svn/wc.db one is actually "ZAP Development Team" ... Does that seem logical?

sgerlach replied:
I think so yes. Does it need some re-organization...?

kingthorin replied:
I think it's just a one line tweak, I'll fire in a PR.
